internal/daemon/controller/handlers/scopes/scope_service.go

```diff
@@ -432,6 +434,76 @@ func (s Service) RotateKeys(ctx context.Context, req *pbs.RotateKeysRequest) (*p
 	return nil, nil
 }
 
+// ListKeyVersionDestructionJobs implements the interface pbs.ScopeServiceServer.
+func (s Service) ListKeyVersionDestructionJobs(ctx context.Context, req *pbs.ListKeyVersionDestructionJobsRequest) (*pbs.ListKeyVersionDestructionJobsResponse, error) {
+	if req.GetScopeId() == "" {
+		req.ScopeId = scope.Global.String()
+	}
+	if err := validateListKeyVersionDestructionJobsRequest(req); err != nil {
+		return nil, err
+	}
+	authResults := s.authResult(ctx, req.GetScopeId(), action.ListScopeKeyVersionDestructionJobs)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+	jobs, err := s.kmsRepo.ListDataKeyVersionDestructionJobs(ctx, req.GetScopeId())
+	if err != nil {
+		return nil, err
+	}
+	res := perms.Resource{
+		Type: resource.Scope,
+	}
+	finalJobs := make([]*pb.KeyVersionDestructionJob, 0, len(jobs))
+	for _, job := range jobs {
+		res.Id = job.ScopeId
+		res.ScopeId = job.ScopeId
+
+		outputFields := authResults.FetchOutputFields(res, action.ListScopeKeyVersionDestructionJobs).SelfOrDefaults(authResults.UserId)
+		outputOpts := make([]handlers.Option, 0, 3)
+		outputOpts = append(outputOpts, handlers.WithOutputFields(&outputFields))
+		if outputFields.Has(globals.ScopeField) {
+			outputOpts = append(outputOpts, handlers.WithScope(authResults.Scope))
+		}
+
+		protoJob, err := destructionJobToProto(ctx, job, outputOpts...)
+		if err != nil {
+			return nil, err
+		}
+		finalJobs = append(finalJobs, protoJob)
+	}
+	return &pbs.ListKeyVersionDestructionJobsResponse{
+		Items: finalJobs,
+	}, nil
+}
+
+// DestroyKeyVersion implements the interface pbs.ScopeServiceServer.
+func (s Service) DestroyKeyVersion(ctx context.Context, req *pbs.DestroyKeyVersionRequest) (*pbs.DestroyKeyVersionResponse, error) {
+	if req.GetScopeId() == "" {
+		req.ScopeId = scope.Global.String()
+	}
+	if err := validateDestroyKeyVersionRequest(req); err != nil {
+		return nil, err
+	}
+	authResults := s.authResult(ctx, req.GetScopeId(), action.DestroyScopeKeyVersion)
+	if authResults.Error != nil {
+		return nil, authResults.Error
+	}
+	destroyed, err := s.kmsRepo.DestroyKeyVersion(ctx, req.GetScopeId(), req.GetKeyVersionId())
+	if err != nil {
+		if errors.Match(errors.T(errors.KeyNotFound), err) {
+			return nil, handlers.NotFoundErrorf("unknown key_version_id %q", req.KeyVersionId)
+		}
+		return nil, err
+	}
+	state := "completed"
+	if !destroyed {
+		state = "pending"
+	}
```

Comment:
I don't think there's much reason to introduce an enum for something as simple as this. It could be a boolean, but this was the API definition we agreed on, which I think is nicer than a boolean and certainly more flexible.

Reply:
I assume by enum you mean something like
```go
type KeyDestructionState int

const (
	KeyDestructionStateUnspecified KeyDestructionState = iota
	KeyDestructionStatePending
	KeyDestructionStateCompleted
)

func (k KeyDestructionState) String() string {
	switch k {
	case KeyDestructionStatePending:
		return "pending"
	case KeyDestructionStateCompleted:
		return "completed"
	default:
		return "unspecified"
	}
}
```

It's quite a bit of overhead for little gain here, I think, and it actually introduces the issue of using the zero value of the type by mistake.